enhancements/big-bundle-unpacking.md

```diff
@@ -0,0 +1,138 @@
+---
+title: big-bundle-unpacking
+authors:
+  - "@njhale"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-08-12
+last-updated: 2020-09-01
+status: implementable
+see-also: - "https://docs.google.com/document/d/1WWbZCyG1sw3vi-1gtMOshmaTaYiMht3cb-Up4NAx4_I/edit#"
+
+---
+
+# Big Bundle Unpacking
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+To ensure Operator Bundles of arbitrary size can be unpacked and used in a Kubernetes cluster by OLM, bundle content will be partitioned across several ConfigMaps. This unpacking behavior will be abstracted behind a new Kubernetes API to allow for future extensibility.
+
+## Motivation
+
+Currently, OLM stages Operator Bundles within a cluster before installation by writing their content into ConfigMaps. There is a 1:1 relationship between bundle and ConfigMap; i.e. each bundle is unpacked to its own ConfigMap. A limitation of this staging method is that the maximum size of a ConfigMap for a given k8s cluster becomes the upper bound for the size of a bundle installed to it.
+
+### Goals
+
+- Enable OLM to install operators from bundles containing more content than fits in a single ConfigMap
+
+### Non-Goals
+
+- Increase the upper storage bound for API resources in general
+- Support bundles containing individual resources that surpass the upper storage bound of a cluster
+
+## Proposal
+
+A new Kubernetes resource kind will be introduced into the `porcelain.operators.coreos.com/v1` group version: `Bundle`. A `Bundle` resource will abstract the pulling and unpacking of an operator bundle from clients (OLM and cluster admins) and will feature the following fields:
```

Comment:
nit: I know porcelain comes from git terminology but I find it very unexpected. Could we use something like `api.operators.coreos.com/v1` instead? Because the `Bundle` resource is effectively an OLM api

Reply:
Is there an actual need to use a separate API group (`porcelain.*`, `api.*`, ...), and not just use `operators.coreos.com/v1` like the rest of the API resources? Is it because it's considered an internal-only API?